tests/phpunit/integration/Core/Consent_Mode/REST_Consent_Mode_ControllerTest.php

```diff
@@ -323,92 +323,92 @@ public function test_get_ads_measurement_status() {
 
 	public function test_get_ads_measurement_status__ads_module_connected() {
 		$this->setup_rest();
-		// Setup the site and admin user to make a successful REST request.
 		$this->grant_manage_options_permission();
 
-		$this->force_module_connection( Ads::MODULE_SLUG );
+		add_filter(
+			'googlesitekit_ads_measurement_connection_checks',
+			function () {
+				return array(
+					function () {
+						return true; },
+				);
+			}
+		);
 
 		$request  = new WP_REST_Request( 'GET', '/' . REST_Routes::REST_ROOT . '/core/site/data/ads-measurement-status' );
 		$response = rest_get_server()->dispatch( $request );
 
-		$response_data = $response->get_data();
-
-		$this->assertTrue( $response_data['connected'] );
+		$this->assertTrue( $response->get_data()['connected'] );
 	}
 
 	public function test_get_ads_measurement_status__ga4_module_connected__ads_connected_setting_is_true() {
 		$this->setup_rest();
-		// Setup the site and admin user to make a successful REST request.
-		$this->grant_manage_options_permission();
-
-		$this->force_module_connection( Analytics_4::MODULE_SLUG );
-
-		$request  = new WP_REST_Request( 'GET', '/' . REST_Routes::REST_ROOT . '/core/site/data/ads-measurement-status' );
-		$response = rest_get_server()->dispatch( $request );
-
-		$response_data = $response->get_data();
-
-		$this->assertFalse( $response_data['connected'] );
-
-		// Set adSenseLinked setting to true, which should mark connection as true.
-		update_option( Analytics_4_Settings::OPTION, array( 'adSenseLinked' => true ) );
-
-		$request  = new WP_REST_Request( 'GET', '/' . REST_Routes::REST_ROOT . '/core/site/data/ads-measurement-status' );
-		$response = rest_get_server()->dispatch( $request );
+			$this->grant_manage_options_permission();
+
+			add_filter(
+				'googlesitekit_ads_measurement_connection_checks',
+				function () {
+					return array(
+						function () {
+							return true; },
+					);
+				}
+			);
```

Comment:
Similar to the above, this test no longer tests what it says it does. We only need to test the handler. These specific conditions should be handled in the respective module tests.